qtconsole/completion_html.py

```diff
@@ -302,7 +304,12 @@ def show_items(self, cursor, items):
             return
         self._start_position = cursor.position()
         self._consecutive_tab = 1
-        items_m, ci = text.compute_item_matrix(items, empty=' ')
+        # Calculate the number of characters available.
+        width = self._text_edit.document().textWidth()
+        char_width = QtGui.QFontMetrics(self._console_widget.font).width(' ')
+        displaywidth = int(max(10, (width / char_width) - 1))
```

Comment:
The width of the terminal in number of characters. In other words: How many characters that can fit into each line in the QTextEdit (i.e. the terminal). The default `displaywidth` used in `text.compute_item_matrix()` is 80, which is why it previously disregarded widget size when laying out the completion suggestions.